src/domain/core/src/services/ingest/data_writer.rs

```diff
@@ -25,6 +26,10 @@ use crate::{AddDataParams, CommitError, OwnedFile};
 /// to commit data into a dataset in bitemporal ledger form.
 #[async_trait::async_trait]
 pub trait DataWriter {
+    /// Returns final output schema of a data slice writer will be appending to
+    /// the dataset, if schema was previously defined.
+    async fn output_schema(&self) -> Option<SchemaRef>;
```

Comment:
Not sure I understand when the result is `None`. Don't we have a schema after the write always?
I mean, if it is not prevoiusly defined and auto-inferred, won't it be returned here?